src/writer/untabber.rs

```diff
@@ -0,0 +1,49 @@
+use std::io::{Result, Write};
+use super::TAB;
+
+pub struct Untabber {
+    orig: Box<Write>,
+}
+
+impl Untabber {
+    pub fn new(orig: Box<Write>) -> Untabber {
+        Untabber{ orig: orig }
+    }
+}
+
+impl Write for Untabber {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        let v = buf.into();
+        match String::from_utf8(v) {
+            Ok(s) => {
+                let bytes = untab(&s).into_bytes();
+                let delta_len = bytes.len() - buf.len();
+                match self.orig.write(&bytes) {
+                    Ok(n) if n < delta_len => Ok(n),
+                    //hack for return normal len
+                    Ok(n) => Ok(n - delta_len),
```

Comment:
In my opinion using `write` is pointless, `write_all` was made specifically to take care of incomplete writes and stuff. There should be a `BufWriter` between this adapter and `File` anyway.